rewrite-java/src/main/java/org/openrewrite/java/cleanup/StringAvoidReplaceAll.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TypeUtils;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * Recipe to avoid using {@link String#replaceAll(String, String)} when the fist argument is not a regular expression.
+ * <p>
+ * The underlying implementation of {@link String#replaceAll(String, String)} calls the {@link Pattern#compile(String)}
+ * method each time it is called even if the first argument is not a regular expression. This has a significant
+ * performance cost and therefore should be used with care.
+ *
+ * @see <a href="https://rules.sonarsource.com/java/RSPEC-5361"></a>
+ * @see <a href="https://docs.oracle.com/javase/tutorial/essential/regex/index.html"></a>
+ */
+public class StringAvoidReplaceAll extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Use `String::replaceAll()` only when fist parameter is a real regular expression";
+    }
+
+    @Override
+    public String getDescription() {
+        return "When `String::replaceAll` is used, the first argument should be a real regular expression. " +
+                "If it’s not the case, `String::replace` does exactly the same thing as `String::replaceAll` without the performance drawback of the regex.";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("RSPEC-5361");
+    }
+
+    @Override
+    public Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(10);
```

Comment:
I looked up the RSPEC and the time estimate is 2min per occurence.

Reply:
Yes, you're right! I changed the value